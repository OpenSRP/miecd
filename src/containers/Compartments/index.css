.compartment-wrapper #breadcrumb {
    color: #0292d3;
}

.compartment-wrapper .breadcrumb-row.row {
    padding-left: 1.5rem;
}

.compartment-wrapper span#breadcrumb-span {
    color: #637381;
    font-size: 1.2rem;
}
#compartment_title {
    font-size: 2.2rem;
}
