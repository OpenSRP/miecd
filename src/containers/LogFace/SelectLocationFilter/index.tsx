/** select dropdown that allows user to filter by location,
 * some constraints applied:
 * - user can only filter for locations that they have been allowed access to,
 * we determine this by using the locationHierarchy of the user returned from security authenticate
 */
import React, { useEffect, useState, Fragment } from 'react';
import { useTranslation } from 'react-i18next';
import Select, { components } from 'react-select';
import { TreeNode } from 'store/ducks/locationHierarchy/types';

interface SelectLocationFilterProps {
    userLocationTree?: TreeNode;
    userLocationId: string;
    disabled: boolean;
    onLocationChange: (value: string) => void;
}

const defaultProps = {
    disabled: false,
};

const SelectLocationFilter = (props: SelectLocationFilterProps) => {
    const { t } = useTranslation();
    const { userLocationTree, userLocationId, disabled, onLocationChange } = props;

    // find the treeNode this user is assigned to
    const userLocationNode = userLocationTree?.first((node) => node.model.id === userLocationId);

    const immediateDescendants: TreeNode[] = userLocationNode?.hasChildren() ? userLocationNode.children : [] ?? [];
    const initialSelectOptions = immediateDescendants.map((node) => ({
        value: node.model.id,
        label: node.model.label,
    }));
    const [options, setOptions] = useState(initialSelectOptions);
    const [hierarchy, setHierarchy] = useState<TreeNode[]>([]);
    const [selectedNode, setSelectedNode] = useState<TreeNode | undefined>();
    const [closeMenuOnSelect, setMenuToClose] = useState<boolean>(false);

    useEffect(() => {
        if (selectedNode) {
            // don't close dropdown after user makes selection if selected node has children
            const hasChildren: boolean = selectedNode.hasChildren();
            setMenuToClose(!hasChildren);
        }
    }, [selectedNode]);

    /** handles a location selection */
    const changeHandler = (value: string) => {
        if (value) {
            // update parentNode and hierarchy
            const thisNode = userLocationTree?.first((node) => node.model.id === value);
            setSelectedNode(thisNode);
            // set options to be children of currently selected node, where selected node is a leaf node, the siblings will be the options
            const nextOptions = (thisNode?.hasChildren() ? thisNode?.children : thisNode?.parent?.children ?? []).map(
                (node: TreeNode) => ({
                    value: node.model.id,
                    label: node.model.label,
                }),
            );
            setOptions(nextOptions);
            const path: TreeNode[] = thisNode?.getPath() as TreeNode[];
            path.pop();
            setHierarchy(path);
        } else {
            //reset node selection-related state variables
            setHierarchy([]);
            setSelectedNode(undefined);
        }

        onLocationChange(value);
    };

    /** Defines filter/search behavior as user types a location name  */
    const customOnInputFilter = (input: string) => {
        if (input) {
            const matchedLocations = userLocationTree?.all((node) =>
                node.model.label.toLowerCase().includes(input.toLowerCase()),
            );
            // create options from the matched locations
            const matchedOptions =
                matchedLocations?.map((node) => {
                    return {
                        value: node.model.id,
                        label: node
                            .getPath()
                            .map((node) => node.model.label)
                            .join(' > '),
                    };
                }) ?? [];

            setOptions(matchedOptions);
        } else {
            if (selectedNode) {
                // set options to be children of currently selected node, where selected node is a leaf node, the siblings will be the options
                const nextOptions = (selectedNode?.hasChildren()
                    ? selectedNode?.children
                    : selectedNode?.parent?.children ?? []
                ).map((node: TreeNode) => ({
                    value: node.model.id,
                    label: node.model.label,
                }));
                setOptions(nextOptions);
            } else {
                setOptions(initialSelectOptions);
            }
        }
    };

    /** render a breadCrumb-ish text showing a hierarchy of the current selected node */
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const Menu = (props: any) => {
        return (
            <Fragment>
                <components.Menu {...props}>
                    <div>
                        {hierarchy.length ? (
                            <div className="hierarchy-actions">
                                <small>
                                    {hierarchy && hierarchy.map((node: TreeNode) => node.model.label).join(' > ')}
                                </small>
                            </div>
                        ) : null}
                        <div>{props.children}</div>
                    </div>
                </components.Menu>
            </Fragment>
        );
    };

    return (
        <Select
            placeholder={t('Select location')}
            options={options}
            components={{ Menu }}
            onChange={(val) => changeHandler(val?.value ?? '')}
            disabled={disabled}
            classNamePrefix={'logface-filters'}
            isSearchable={true}
            isClearable={true}
            onInputChange={customOnInputFilter}
            closeMenuOnSelect={closeMenuOnSelect}
        />
    );
};

SelectLocationFilter.defaultProps = defaultProps;

export { SelectLocationFilter };
