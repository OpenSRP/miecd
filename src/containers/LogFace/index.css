thead {
    width: 100%;
    position: absolute;
    display: table;
}

tbody {
    display: table;
    width: 100%;
}

thead tr {
    border-bottom: 0;
}

tbody tr {
    align-items: center;
}

td {
    text-align: left;
    border: 1px solid rgba(0, 0, 0, 0.2);
    padding: 5px;
}

.table-container-div {
    box-shadow: 0 2px 4px 0 #d7d7e0;
    display: block;
}

.table-container-div table tbody tr td:first-child {
    overflow-wrap: break-word;
}

.table-container-div table {
    border: none !important;
    background: #fff;
}

.table-container-div .card {
    margin-bottom: 16px;
}

.table-container-div table th {
    color: #3a484f;
    font-family: Roboto;
    font-size: 16px;
    font-weight: 400;
}

.table-container-div table td {
    color: #7c7c7c;
    font-family: Roboto;
    font-size: 18px;
    font-weight: 400;
}

.filter-panel {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: flex-start;
}

.logface-content .filter-panel .filters {
    display: flex;
    justify-content: flex-start;
    flex-grow: 2;
    padding: 0px 0px 10px;
}

.logface-content .filter-panel .filters .logface-page-filter {
    display: block;
    padding: 10px 20px;
}

.form-group {
    vertical-align: bottom;
}

.form-control {
    margin-top: 2%;
    width: auto;
}

#searchbox {
    display: flex;
    align-items: center;
    justify-content: center;
    padding-top: 20px;
}

#logface_title {
    padding: 0px 15px;
    border-left: 5px solid #0093db;
    color: #001429;
    font-family: Roboto;
    font-size: 30px;
    letter-spacing: -0.18px;
    line-height: 34px;
    margin-top: 24px;
}

.logface-content .table-container-div table .badge-warning {
    color: #fff;
}

.logface-content {
    padding-left: 24px;
    padding-right: 48px;
}

#header {
    position: relative;
    top: 0;
    height: 50px;
    border-bottom: 0.5px solid rgba(0, 0, 0, 0.2);
}

.table-container-div table #body {
    overflow: auto;
    height: 100%;
}

.table-container-div table #body > tr td:nth-child(8) {
    width: 26%;
    height: 20%;
}

.large-width {
    width: 26%;
}

.default-width {
    width: 12%;
}

.small-width {
    width: 7%;
}

.paginator {
    display: flex;
    justify-content: center;
}

.table-striped tbody tr:nth-of-type(odd) {
    background-color: #fff !important;
}

.table-striped tbody tr:nth-of-type(even) {
    background-color: rgba(117, 117, 117, 0.05) !important;
}

#export-button {
    align-self: center;
    margin-left: auto;
    margin-top: 1rem;
    background: #0093db;
    color: #fff;
    border-radius: 3px;
    text-decoration: none;
    padding: 5px 15px;
    font-family: Roboto;
    font-size: 16px;
}

#export-button:hover {
    background: #0189cb;
}

.filters .logface-page-filter .dropdown button {
    background-color: #ffffff;
    border: 0.5px solid #bababa;
    border-radius: 2px;
    box-shadow: 0 1px 3px 1px rgba(127, 131, 155, 0.15), 0 1px 2px 0 rgba(127, 131, 155, 0.3);
    height: 40px;
    font-family: Roboto;
    font-size: 14px;
    font-weight: 500;
    line-height: 16px;
    text-align: left;
}

.dropdown button span {
    padding-right: 35px;
    color: #666666;
    font-size: 14px;
    font-weight: 500;
}

.filters .logface-page-filter span {
    color: #212121;
    font-family: Roboto;
    font-size: 18px;
    font-weight: 400;
    line-height: 17px;
    opacity: 0.8;
}

.filter-panel input.logface-search {
    border-radius: 0;
    margin-top: 35px;
    min-width: 200px;
    max-width: 300px;
    background: url(/src/assets/images/search-icon.svg) no-repeat;
    background-position: right center;
    background-color: white;
    padding-left: 0.2rem;
}

.filter-panel input.logface-search:focus {
    box-shadow: 0 1px 3px 1px rgba(127, 131, 155, 0.15), 0 1px 2px 0 rgba(127, 131, 155, 0.3);
}

.filter-panel input.logface-search::placeholder {
    color: #c2c2c2;
    font-family: Roboto;
    font-size: 15px;
}

.btn-secondary {
    color: #676767;
    border: 0.5px solid #bababa;
    box-shadow: #7f839b;
    background-color: #fff;
}

#next,
#previous {
    background: #0093db;
    color: #fff;
    border-radius: 2px;
    width: 120px;
    height: 32px;
}

.logface-filters__control {
    min-width: 200px;
    border-radius: 0 !important;
}

.logface-filters__placeholder {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.hierarchy-actions {
    padding: 5px;
}

.pagination-container {
    display: flex;
    justify-content: center;
}
