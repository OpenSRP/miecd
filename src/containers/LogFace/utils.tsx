import { replace, split, trim } from 'lodash';
import React from 'react';

/** convert the smsData message field from prose to more easily readable
 * point format
 *
 * @param message - the message prose.
 */
export const parseMessage = (message: string) => {
    const propValues = split(message, '\n');
    const replacedEquals = propValues.map(trim).map((entry) => replace(entry, / =\s*/, ' : '));
    return (
        <ul>
            {replacedEquals.map((value, index) => {
                return <li key={`${value}-${index}`}>{value}</li>;
            })}
        </ul>
    );
};
