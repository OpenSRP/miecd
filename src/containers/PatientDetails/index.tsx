/* eslint-disable @typescript-eslint/no-use-before-define */
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { TFunction } from 'i18next';
import { useTranslation } from 'react-i18next';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { useLastLocation } from 'react-router-last-location';
import { Store } from 'redux';
import BasicInformation, { LabelValuePair } from '../../components/BasicInformation';
import { BACKPAGE_ICON, COMMUNE, DISTRICT, MODULE_SEARCH_PARAM_KEY, PROVINCE, VILLAGE } from '../../constants';
import * as React from 'react';
import './index.css';
import { flatten, keyBy } from 'lodash';
import { fetchData, fetchSupersetData, sortByEventDate, useHandleBrokenPage } from 'helpers/utils';
import supersetFetch from '../../services/superset';
import { ErrorPage } from 'components/ErrorPage';
import Ripple from 'components/page/Loading';
import { getLocationsOfLevel, Location } from '../../store/ducks/locations';
import locationsReducer, { reducerName as locationReducerName } from '../../store/ducks/locations';
import {
    CompartmentSmsTypes,
    getSmsDataByFilters,
    LogFaceSmsType,
    NbcPncSmsData,
    NutritionSmsData,
    PregnancySmsData,
} from '../../store/ducks/sms_events';
import smsReducer, { reducerName as smsReducerName } from '../../store/ducks/sms_events';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { PatientDetailsReport } from 'components/PatientDetailsReports';
import { parse } from 'query-string';
import { LogFaceModules } from 'configs/settings';
import { ConnectedChildChart } from 'components/PatientDetailsCharts/ChildChart';
import { ConnectedMotherChart } from 'components/PatientDetailsCharts/MotherChart';
import { useQueries } from 'react-query';
import {
    COMPARTMENTS_NBC_AND_PNC_SLICE,
    COMPARTMENTS_NUTRITION_SLICE,
    COMPARTMENTS_PREGNANCY_SLICE,
} from 'configs/env';
import superset from '@onaio/superset-connector';

reducerRegistry.register(smsReducerName, smsReducer);
reducerRegistry.register(locationReducerName, locationsReducer);

interface RouteParams {
    patient_id: string;
}

interface PatientDetailProps extends RouteComponentProps<RouteParams> {
    isChild: boolean;
    provinces: Location[];
    districts: Location[];
    communes: Location[];
    villages: Location[];
    supersetService: typeof supersetFetch;
    logFaceReports: LogFaceSmsType[];
}

const defaultProps = {
    isChild: false,
    provinces: [],
    districts: [],
    communes: [],
    villages: [],
    supersetService: supersetFetch,
    logFaceReports: [],
};

const PatientDetails = (props: PatientDetailProps) => {
    const { isChild, communes, villages, districts, provinces, supersetService, logFaceReports } = props;
    const [loading, setLoading] = React.useState(true);
    const { error, handleBrokenPage, broken } = useHandleBrokenPage();
    const { t } = useTranslation();
    const lastLocation = useLastLocation();

    const { patient_id: patientId } = props.match.params;

    React.useEffect(() => {
        fetchData(supersetService)
            .catch((err) => {
                handleBrokenPage(err);
            })
            .finally(() => setLoading(false));
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const supersetFilterOptions = superset.getFormData(
        1,
        [
            {
                comparator: patientId,
                operator: '==',
                subject: 'anc_id',
            },
        ],
        { event_date: false },
    );

    const basicInfoQueries = useQueries(
        [COMPARTMENTS_NBC_AND_PNC_SLICE, COMPARTMENTS_NUTRITION_SLICE, COMPARTMENTS_PREGNANCY_SLICE].map((slice) => {
            return {
                queryKey: [slice, patientId],
                queryFn: () => fetchSupersetData<CompartmentSmsTypes>(slice, t, supersetFilterOptions, supersetService),
            };
        }),
    );

    const mostRecentReports = flatten(basicInfoQueries.map((query) => query.data)) as CompartmentSmsTypes[];

    if (loading) {
        return <Ripple />;
    }

    if (broken) {
        return <ErrorPage title={error?.name} message={error?.message} />;
    }

    const basicInformationValuePairs = getBasicInformationProps(
        patientId,
        isChild,
        mostRecentReports,
        t,
        communes,
        districts,
        villages,
        provinces,
    );

    return (
        <div className="patient-details">
            <Link to={lastLocation ? lastLocation.pathname : '#'} className="back-page">
                <FontAwesomeIcon icon={BACKPAGE_ICON} size="lg" />
                <span>{t('Back')}</span>
            </Link>
            <div id="titleDiv">
                <h2 id="patients_title">{t('Patient Details')}</h2>
            </div>
            <BasicInformation labelValuePairs={basicInformationValuePairs} />

            <PatientDetailsReport patientsReports={logFaceReports} isChild={isChild}></PatientDetailsReport>

            {isChild ? <ConnectedChildChart {...props} /> : <ConnectedMotherChart {...props} />}
        </div>
    );
};

function getBasicInformationProps(
    patientId: string,
    isChild: boolean,
    recentSmsData: CompartmentSmsTypes[],
    t: TFunction,
    communes: Location[],
    districts: Location[],
    villages: Location[],
    provinces: Location[],
): LabelValuePair[] {
    const defaultNA = t('N/A');
    const defaultAge = defaultNA;
    const mostRecentReport = sortByEventDate(recentSmsData)[0] ?? {};

    const {
        lmp_edd: edd,
        event_id: smsId,
        age,
        gravidity,
        parity,
        risk_level: motherRiskCat,
        health_insurance: healthInsuranceNum,
        height,
        weight,
        handwashing: handWashing,
        household,
        toilet,
        ethnicity,
        planned_delivery_location: deliveryLocation,
        previous_risks: previousRisks,
        mother_symptoms: motherSymptoms,
        child_symptoms: childSymptoms,
    } = mostRecentReport as NutritionSmsData & PregnancySmsData & NbcPncSmsData;

    const locationPath = getPathFromSupersetLocs(
        provinces,
        districts,
        communes,
        villages,
        mostRecentReport.location_id,
    );
    const location = locationPath.map((location) => location.location_name).join(', ');
    const commonLabels = [
        { label: t('SMS ID'), value: smsId },
        { label: t('Patient ID'), value: patientId },
        { label: t('Age'), value: age ?? defaultAge },
        { label: t('Location'), value: location ?? defaultNA },
        { label: t('Current Weight'), value: weight ?? defaultNA },
        { label: t('Current Height'), value: height ?? defaultNA },
        { label: t('Health Insurance number'), value: healthInsuranceNum ?? defaultNA },
        { label: t('Household Type'), value: household ?? defaultNA },
        { label: t('Ethnicity'), value: ethnicity ?? defaultNA },
        { label: t('HandWashing'), value: handWashing ?? defaultNA },
        { label: t('Toilet'), value: toilet ?? defaultNA },
    ];
    const basicInformationProps = !isChild
        ? ([
              ...commonLabels,
              { label: t('Current Parity'), value: parity ?? defaultNA },
              { label: t('Current Gravidity'), value: gravidity ?? defaultNA },
              { label: t('Current EDD'), value: edd ?? defaultNA },
              { label: t('Current Risk categorization'), value: motherRiskCat ?? defaultNA },
              { label: t('Delivery plan/ Location of delivery'), value: deliveryLocation ?? defaultNA },
              { label: t('Previous risks and Existing conditions'), value: previousRisks ?? defaultNA },
              { label: t('Current risks'), value: motherSymptoms ?? defaultNA },
          ] as LabelValuePair[])
        : ([...commonLabels, { label: t('Current risks'), value: childSymptoms ?? defaultNA }] as LabelValuePair[]);
    return basicInformationProps;
}

/** returns location path starting with the smallest level i.e village to province */
export const getPathFromSupersetLocs = (
    provinces: Location[],
    district: Location[],
    commune: Location[],
    village: Location[],
    location_id: string,
) => {
    // TODO this locations do not need be different locations arrays
    const provincesById = keyBy(provinces, (location) => location.location_id);
    const districtById = keyBy(district, (location) => location.location_id);
    const communeByID = keyBy(commune, (location) => location.location_id);
    const villageById = keyBy(village, (location) => location.location_id);

    const path = [];
    const thisVillage = villageById[location_id];
    path.push(thisVillage);
    const thisCommune = communeByID[thisVillage?.parent_id];
    path.push(thisCommune);
    const thisDistrict = districtById[thisCommune?.parent_id];
    path.push(thisDistrict);
    const thisProvince = provincesById[thisDistrict?.parent_id];
    path.push(thisProvince);
    return path.filter((location) => !!location);
};

PatientDetails.defaultProps = defaultProps;
export { PatientDetails };

const smsByFilters = getSmsDataByFilters();

type MapStateToProps = Pick<PatientDetailProps, 'communes' | 'districts' | 'provinces' | 'villages' | 'logFaceReports'>;
const mapStateToProps = (state: Partial<Store>, ownProps: PatientDetailProps): MapStateToProps => {
    const patientId = ownProps.match.params.patient_id;

    const module = parse(ownProps.location.search)[MODULE_SEARCH_PARAM_KEY] as LogFaceModules | undefined;

    const logFaceReports = smsByFilters(state, {
        patientId: patientId,
        module,
    });

    return {
        communes: getLocationsOfLevel(state, COMMUNE),
        districts: getLocationsOfLevel(state, DISTRICT),
        provinces: getLocationsOfLevel(state, PROVINCE),
        villages: getLocationsOfLevel(state, VILLAGE),
        logFaceReports,
    };
};

const ConnectedPatientDetails = connect(mapStateToProps, null)(PatientDetails);

export default ConnectedPatientDetails;
