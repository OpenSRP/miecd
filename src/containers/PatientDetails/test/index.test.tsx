import reducerRegistry from '@onaio/redux-reducer-registry';
import { ConnectedRouter } from 'connected-react-router';
import { PATIENT_DETAIL_URL, PREGNANCY, PREGNANCY_LOGFACE_URL } from '../../../constants';
import { LogFacePropsType } from 'containers/LogFace';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, RouteComponentProps, Router } from 'react-router';
import { clearLocationSlice } from 'store/ducks/locations';
import { mountWithTranslations } from '../../../helpers/testUtils';
import smsReducer, { reducerName, removeSms } from '../../../store/ducks/sms_events';
import store from '../../../store/index';
import ConnectedPatientDetails from '..';
import { smsDataFixture } from 'store/ducks/tests/fixtures';
import * as securityAuthenticate from '../../../store/ducks/tests/fixtures/securityAuthenticate.json';
import { communes, districts, provinces, villages } from '../../HierarchichalDataTable/test/fixtures';
import { userLocations } from 'containers/LogFace/tests/userLocationFixtures';
import { authenticateUser } from '@onaio/session-reducer';

// /** register the superset reducer */
reducerRegistry.register(reducerName, smsReducer);

const history = createBrowserHistory();

// eslint-disable-next-line @typescript-eslint/no-var-requires
const fetch = require('jest-fetch-mock');

jest.mock('../../../configs/env');

jest.mock('highcharts');

const locationProps = {
    history,
    location: {
        hash: '',
        pathname: `${PATIENT_DETAIL_URL}`,
        search: '',
        state: {},
    },
    match: {
        isExact: true,
        params: { patient_id: '' },
        path: `${PATIENT_DETAIL_URL}`,
        url: `${PATIENT_DETAIL_URL}`,
    },
};

describe('containers/LogFace extended', () => {
    beforeAll(() => {
        store.dispatch(
            authenticateUser(
                true,
                {
                    email: 'bob@example.com',
                    name: 'Bobbie',
                    username: 'RobertBaratheon',
                },
                { api_token: 'hunter2', oAuth2Data: { access_token: 'hunter2', state: 'abcde' } },
            ),
        );
    });

    beforeEach(() => {
        jest.resetAllMocks();
        store.dispatch(clearLocationSlice());
        store.dispatch(removeSms);
        fetch.resetMocks();
    });

    it('shows loader and Breaks just fine', async () => {
        const supersetFetchMock = jest.fn(async () => []);
        fetch.mockReject(new Error('coughid'));
        const props = {
            ...locationProps,
            supersetService: supersetFetchMock,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <ConnectedRouter history={history}>
                    <ConnectedPatientDetails {...props} />
                </ConnectedRouter>
            </Provider>,
        );

        expect(wrapper.find('Ripple')).toHaveLength(1);

        await act(async () => {
            await new Promise((resolve) => setImmediate(resolve));
            wrapper.update();
        });

        // loader is no longer showing
        expect(wrapper.find('Ripple')).toHaveLength(0);

        // showing broken page due to error
        expect(wrapper.text()).toMatchInlineSnapshot(`"An error occurredErrorcoughid"`);
        // and finally check the requests made
        expect(supersetFetchMock.mock.calls).toEqual([
            ['userLocation'],
            ['province'],
            ['district'],
            ['commune'],
            ['village'],
            ['smsData'],
        ]);
        expect(fetch.mock.calls).toEqual([
            [
                'https://test.smartregister.org/opensrp/rest//security/authenticate/',
                {
                    headers: {
                        accept: 'application/json',
                        authorization: 'Bearer hunter2',
                        'content-type': 'application/json;charset=UTF-8',
                    },
                    method: 'GET',
                },
            ],
        ]);

        wrapper.unmount();
    });

    it('renders correctly', async () => {
        const supersetFetchMock = jest
            .fn()
            .mockResolvedValueOnce(userLocations)
            .mockResolvedValueOnce(provinces)
            .mockResolvedValueOnce(districts)
            .mockResolvedValueOnce(communes)
            .mockResolvedValueOnce(villages)
            .mockResolvedValueOnce(smsDataFixture);
        fetch.mockResponse(JSON.stringify(securityAuthenticate));
        const props = {
            ...locationProps,
            supersetService: supersetFetchMock,
        };
        const container = document.createElement('div');
        document.body.appendChild(container);

        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <ConnectedRouter history={history}>
                    <ConnectedPatientDetails {...props} />
                </ConnectedRouter>
            </Provider>,
        );

        expect(wrapper.find('Ripple')).toHaveLength(1);

        await act(async () => {
            await new Promise((resolve) => setImmediate(resolve));
            wrapper.update();
        });

        wrapper.find('.information__card_body_cell').forEach((cell) => {
            expect(toJson(cell)).toMatchSnapshot('Basic Info key value');
        });

        expect(wrapper.find('ReportTable')).toHaveLength(1);

        expect(toJson(wrapper.find('#titleDiv'))).toMatchSnapshot('titleDiv');

        wrapper.unmount();
    });
});
