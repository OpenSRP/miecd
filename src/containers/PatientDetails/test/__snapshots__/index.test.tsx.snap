// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/LogFace extended renders correctly: Basic Info key value 1`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Name-Runny  NULL Danny"
>
  <span
    className="prop"
  >
     
    Name
    :
  </span>
  <span
    className="value"
  >
    Runny  NULL Danny
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 2`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Patient ID-"
>
  <span
    className="prop"
  >
     
    Patient ID
    :
  </span>
  <span
    className="value"
  />
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 3`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Age-N/A"
>
  <span
    className="prop"
  >
     
    Age
    :
  </span>
  <span
    className="value"
  >
    N/A
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 4`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Location-"
>
  <span
    className="prop"
  >
     
    Location
    :
  </span>
  <span
    className="value"
  />
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 5`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Current Weight-N/A"
>
  <span
    className="prop"
  >
     
    Current Weight
    :
  </span>
  <span
    className="value"
  >
    N/A
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 6`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Current Height-N/A"
>
  <span
    className="prop"
  >
     
    Current Height
    :
  </span>
  <span
    className="value"
  >
    N/A
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 7`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Health Insurance number- NA"
>
  <span
    className="prop"
  >
     
    Health Insurance number
    :
  </span>
  <span
    className="value"
  >
     NA
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 8`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Household Type-  NULL"
>
  <span
    className="prop"
  >
     
    Household Type
    :
  </span>
  <span
    className="value"
  >
      NULL
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 9`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Ethnicity-  NULL"
>
  <span
    className="prop"
  >
     
    Ethnicity
    :
  </span>
  <span
    className="value"
  >
      NULL
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 10`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="HandWashing-  NULL"
>
  <span
    className="prop"
  >
     
    HandWashing
    :
  </span>
  <span
    className="value"
  >
      NULL
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 11`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Toilet-  NULL"
>
  <span
    className="prop"
  >
     
    Toilet
    :
  </span>
  <span
    className="value"
  >
      NULL
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 12`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Current Blood Pressure-  NULL"
>
  <span
    className="prop"
  >
     
    Current Blood Pressure
    :
  </span>
  <span
    className="value"
  >
      NULL
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 13`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Current Gravidity-N/A"
>
  <span
    className="prop"
  >
     
    Current Gravidity
    :
  </span>
  <span
    className="value"
  >
    N/A
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 14`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Current EDD-30y "
>
  <span
    className="prop"
  >
     
    Current EDD
    :
  </span>
  <span
    className="value"
  >
    30y 
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 15`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Current Parity-N/A"
>
  <span
    className="prop"
  >
     
    Current Parity
    :
  </span>
  <span
    className="value"
  >
    N/A
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: Basic Info key value 16`] = `
<div
  className="flex-fill d-flex flex-wrap information__card_body_cell p-1"
  key="Previous Pregnancy Risk- NULL"
>
  <span
    className="prop"
  >
     
    Previous Pregnancy Risk
    :
  </span>
  <span
    className="value"
  >
     NULL
  </span>
</div>
`;

exports[`containers/LogFace extended renders correctly: titleDiv 1`] = `
<div
  id="titleDiv"
>
  <h2
    id="patients_title"
  >
    Patient Details
  </h2>
</div>
`;
